lib/middleware/verify_hideout_id.rb

```diff
@@ -0,0 +1,39 @@
+require 'json'
+
+require_relative '../../app/helpers/auth_helper.rb'
+
+module Middleware
+    class VerifyHideoutId
+      def initialize(app)
+        @app = app
+      end
+  
+      def call(env)
+        request = ActionDispatch::Request.new(env)
+        request_identifier = "#{request.method} #{request.path}"
+        if not ENV['PUBLIC_ROUTES'].include?(request_identifier)
+          authoriation_header = request.headers[:Authorization]
+          return [401, {}, []] if authoriation_header.nil?
+  
+          access_token = authoriation_header.split[1]
+          result = AuthHelper.validate_token_by_type(:ACCESS, access_token)
+          return [401, {}, []] unless result[:success]
+  
+          begin
+            payload = result[:payload]
+            payload.transform_keys!(&:to_sym)
+            current_hideout = Hideout.find_by(id: payload[:hideout_id])
+            return [400, {}, []] unless current_hideout == request.params[:hideout_id].to_i
```

Comment:
Ok let's not check like this. I think we should move the lines from the access token middleware that find and add the hideout_id to the request params into this middleware. So you don't need to do any checks like this since `find_by!` will automatically return an error if the Hideout isn't found. Since we will move the lines from the other middleware to here, you won't have access to the hideout_id from the params and payload, instead retrieve it from the request.params itself where it's specified in the url.